tests/test_actions.py

```diff
@@ -1097,3 +1097,79 @@ def test_get_cs_status_raises_unexpected_exceptions(
         )
         with pytest.raises(ClientError):
             self.actions._get_cs_status(sentinel.change_set_name)
+
+    def test_fetch_remote_template_no_template(self):
+        self.actions.connection_manager.call.side_effect = ClientError(
+            {
+                "Error": {
+                    "Code": "ValidationError",
+                    "Message": "An error occurred (ValidationError) "
+                               "when calling the GetTemplate operation: "
+                               "Stack with id foo does not exist"
+                }
+            },
+            sentinel.operation
+        )
+
+        with pytest.raises(ClientError):
+            self.actions.fetch_remote_template()
+
+    @patch("sceptre.plan.actions.StackActions.fetch_remote_template")
```

Comment:
I don't expect these tests to pass at the moment. These were pulled over from @alexharv074 's branch. Once we can agree on a way forward, I'll fix them all up and add more.